packages/graphqlgen/src/tests/typescript/__snapshots__/basic.test.ts.snap

```diff
@@ -0,0 +1,342 @@
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type DimensionsResolver = (
+    parent: Image,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => Dimensions | Promise<Dimensions>;
+
+  export interface Type {
+    id: (
+      parent: Image,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    url: (
+      parent: Image,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    dimensions: (
+      parent: Image,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => Dimensions | Promise<Dimensions>;
+  }
+}
+
+export namespace VideoResolvers {
+  export const defaultResolvers = {};
+
+  export type IdResolver = (
+    parent: Video,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type UrlResolver = (
+    parent: Video,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => string | Promise<string>;
+
+  export type DurationResolver = (
+    parent: Video,
+    args: {},
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => number | Promise<number>;
+
+  export interface Type {
+    id: (
+      parent: Video,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    url: (
+      parent: Video,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => string | Promise<string>;
+
+    duration: (
+      parent: Video,
+      args: {},
+      ctx: Context,
+      info: GraphQLResolveInfo
+    ) => number | Promise<number>;
+  }
+}
+
+export interface Resolvers {
+  Query: QueryResolvers.Type<T>;
+  Dimensions: DimensionsResolvers.Type<T>;
+  Image: ImageResolvers.Type<T>;
+  Video: VideoResolvers.Type<T>;
+  Media?: MediaResolvers.Type<T>;
+}
+"
+`;
+
+exports[`basic union: generatedTypes 1`] = `
+"// Code generated by github.com/prisma/graphqlgen, DO NOT EDIT.
+
+import {
+  GraphQLResolveInfo,
+  GraphQLTypeResolver,
+  GraphQLIsTypeOfFn
+} from \\"graphql\\";
+import { Context } from \\"../../../fixtures/union/types\\";
+import { User } from \\"../../../fixtures/union/types\\";
+import { UserType } from \\"../../../fixtures/union/types\\";
+import { Student } from \\"../../../fixtures/union/types\\";
+import { Professor } from \\"../../../fixtures/union/types\\";
+
+export namespace UserTypeResolvers {
+  export interface Type {
+    __resolveType?: GraphQLTypeResolver<Student | Professor, Context>;
+    __isTypeOf?: GraphQLIsTypeOfFn<Student | Professor, Context>;
+  }
+}
+
+export namespace QueryResolvers {
+  export const defaultResolvers = {};
+
+  export interface ArgsUsers {
+    first: number | null;
+  }
+
+  export type UsersResolver = (
+    parent: {},
+    args: ArgsUsers,
+    ctx: Context,
+    info: GraphQLResolveInfo
+  ) => UserType[] | Promise<UserType[]>;
```

Comment:
I'm not sure on the generated type here. Might be better to return `Array<Student | Professor> | Promise<Array<Student | Professor>>`, which is the case for interfaces.